[Khoo Ming Wei, Benedict] iP

benedictkhoomw opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

benedictkhoomw committed:
Add greet, echo and exit functionality

benedictkhoomw committed:
Add level 2 functionality

benedictkhoomw committed:
Add Level 3 functionality

benedictkhoomw committed:
Extract list and done commands to methods

benedictkhoomw committed:
Restructure command parsing and add level 4 functionality

Add Deadline subclass of Task
Add Event subclass of Task
Add ToDo subclass of Task
Replace all commands parsing (except exit) with regex
Extract command parsing and handling to separate method

benedictkhoomw committed:
Fix done command invalid argument message

benedictkhoomw committed:
Add Text UI testing functionality

benedictkhoomw committed:
Add error handling

benedictkhoomw committed:
Fix done command regex and improve task number out of range message

benedictkhoomw committed:
Add tests for Level-5

benedictkhoomw committed:
Fix event error message

benedictkhoomw committed:
Add delete functionality and tests

benedictkhoomw committed:
Add save functionality

Let's store the user's tasks upon update so that later we can load
them. This way the user can track tasks across multiple sessions.

benedictkhoomw committed:
Fix style of Duke.java

benedictkhoomw committed:
Add load functionality

Let's load the saved tasks on program start so the user can view their
tasks from the previous session

benedictkhoomw committed:
Add response when listing empty task list

benedictkhoomw committed:
Add data/ to gitignore

benedictkhoomw committed:
Add date recognition to deadline command

Format recognized: yyyy-mm-dd

benedictkhoomw committed:
Merge branch 'branch-Level-7'

benedictkhoomw committed:
Merge branch 'branch-Level-8'

benedictkhoomw committed:
Refactor Duke to use more OOP

benedictkhoomw committed:
Format code

benedictkhoomw committed:
Organize code into packages

benedictkhoomw committed:
Change source folder structure

From src/duke/ to src/main/java/duke/

benedictkhoomw committed:
Add tests for Parser and ToDo

benedictkhoomw committed:
Add javadocs for all public classes and methods

benedictkhoomw committed:
Add case-sensitive search functionality

benedictkhoomw committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-CodingStandard

benedictkhoomw committed:
Fix import order

benedictkhoomw committed:
Fix ParserTest missing @Test

benedictkhoomw committed:
Merge branch 'branch-A-CodingStandard'

benedictkhoomw committed:
Merge branch 'branch-A-JavaDoc'

benedictkhoomw committed:
Merge branch 'branch-Level-9'

benedictkhoomw committed:
Fix FindCommand import order

benedictkhoomw committed:
Change welcome logo

Reason: checkstyle was detecting lines longer than 120 chars

benedictkhoomw committed:
Fix ParserTest and ToDoTest import order

SoonKeatNeo commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,71 @@
+import duke.ui.Ui;
+
+/**
+ * A personal task list app.
```
Perhaps following the header comment format in JavaDocs style would be nice?

SoonKeatNeo commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,71 @@
+import duke.ui.Ui;
+
+/**
+ * A personal task list app.
```
Perhaps adding author and version tags to the javadocs may be good as well?

benedictkhoomw replied:
Thank you for the suggestion. I have added author tags to all my source files.

SoonKeatNeo commented on src/main/java/duke/commands/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,90 @@
+public class AddDeadlineCommand extends Command {
+    private static final Pattern ADD_DEADLINE_KEYWORD = Pattern.compile("(?i)deadline\\b");
+    private static final Pattern ADD_DEADLINE_DESC = Pattern.compile("(?i)deadline\\s+(\\w.*)");
+    private static final Pattern ADD_DEADLINE_DATE = Pattern.compile("(?i)deadline\\s+(\\w.*)\\s+/by\\s+(\\w.*)");
```
I like the clean regex used to filter commands!

SoonKeatNeo commented on src/main/java/duke/task/ToDo.java:
```diff
@@ -0,0 +1,92 @@
+public class ToDo extends Task {
+    private static final String TYPE = "T";
+    private static final String SEPARATOR = "\\|";
+    private static final int FIELD_COUNT = 3;
```
I also like that you also declared constants to prevent numbers from appearing in the code with no context later!

SoonKeatNeo reviewed (commented):
I am glad I had a chance to review your PR, since your code quality is observably much higher than mine! I've learnt from your methods of abstracting out the commands and I appreciate it!

I like that everything is well-commented, and generally your code quality appears to be pretty high as well! There's clean regex for your parsing and filtering and it looks really good 👍

benedictkhoomw committed:
Add JavaFX GUI

benedictkhoomw committed:
Add Javadocs for GUI files, and reorder imports

benedictkhoomw committed:
Merge branch 'branch-Level-10'

benedictkhoomw committed:
Add @author tag to all source files and format Javadocs

benedictkhoomw committed:
Add formatting and styling to GUI

benedictkhoomw committed:
Merge branch 'branch-Level-10'

creativequotient commented on src/main/java/duke/Main.java:
```diff
@@ -0,0 +1,19 @@
+ *
+ * @author Benedict Khoo
+ */
+public class Main {
+    /**
+     * Launches the Duke application.
+     *
+     * @param args Ignored command line arguments.
+     */
+    public static void main(String[] args) {
+        Application.launch(Duke.class, args);
+    }
+}
```
I like how clean your driver class is for the entire program!

creativequotient commented on src/main/java/duke/commands/AddEventCommand.java:
```diff
@@ -0,0 +1,84 @@
+ *
+ * @author Benedict Khoo
+ */
+public class AddEventCommand extends Command {
+    private static final Pattern ADD_EVENT_KEYWORD = Pattern.compile("(?i)event\\b");
+    private static final Pattern ADD_EVENT_DESC = Pattern.compile("(?i)event\\s+(\\w.*)");
+    private static final Pattern ADD_EVENT_PERIOD = Pattern.compile("(?i)event\\s+(\\w.*)\\s+/at\\s+(\\w.*)");
```
I like the use of regex to parse the input command!

creativequotient commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,55 @@
+    public Command parseCmd(String rawInput) {
+        String cmd = rawInput.trim();
+
+        try {
+            if (ExitCommand.isExitCommand(cmd)) {
+                return new ExitCommand();
+            } else if (AddToDoCommand.isAddToDoCommand(cmd)) {
+                return AddToDoCommand.parseAddToDoCommand(cmd);
+            } else if (AddDeadlineCommand.isAddDeadlineCommand(cmd)) {
+                return AddDeadlineCommand.parseAddDeadlineCommand(cmd);
+            } else if (AddEventCommand.isAddEventCommand(cmd)) {
+                return AddEventCommand.parseAddEventCommand(cmd);
+            } else if (ListCommand.isListCommand(cmd)) {
+                return ListCommand.parseListCommand(cmd);
+            } else if (DoneCommand.isDoneCommand(cmd)) {
+                return DoneCommand.parseDoneCommand(cmd);
+            } else if (DeleteCommand.isDeleteCommand(cmd)) {
+                return DeleteCommand.parseDeleteCommand(cmd);
+            } else if (FindCommand.isFindCommand(cmd)) {
+                return FindCommand.parseFindCommand(cmd);
+            } else {
+                return new InvalidCommand(String.format("Sorry, I don't know what '%s' means", cmd));
+            }
+        } catch (DukeException ex) {
+            return new InvalidCommand(ex.getMessage());
+        }
```
Perhaps a switch statement would be suitable here! :)

creativequotient commented on src/main/java/duke/ui/Gui.java:
```diff
@@ -0,0 +1,166 @@
+    @Override
+    public void showGreeting() {
+        dialogContainer.getChildren().add(
+                DialogBox.getDukeDialog("Hello, I am Ying Jen.\n"
```
I like that you've given your bot a name!

creativequotient reviewed (commented):
It was a pleasure to review you code. I particularly liked your use of regex to parse the input commands!

benedictkhoomw committed:
Remove debug print code in Gui.java

benedictkhoomw committed:
Update mainClassName in build.gradle

benedictkhoomw committed:
Add null check assertions

benedictkhoomw committed:
Merge pull request #1 from benedictkhoomw/branch-A-Assertions

Add null check assertions

benedictkhoomw committed:
Refactor code to adhere to code quality guidelines

benedictkhoomw committed:
Refactor Gui.handleUserInput to lambda

benedictkhoomw committed:
Refactor TaskList.serialize to use streams

benedictkhoomw committed:
Merge pull request #2 from benedictkhoomw/branch-A-CodeQuality

Refactor code to adhere to code quality guidelines

benedictkhoomw committed:
Merge branch 'master' into branch-A-Lambdas

benedictkhoomw committed:
Merge branch 'master' into branch-A-Streams

benedictkhoomw committed:
Merge pull request #4 from benedictkhoomw/branch-A-Streams

Refactor TaskList serialization to use streams

benedictkhoomw committed:
Merge branch 'master' into branch-A-Lambdas

benedictkhoomw committed:
Merge pull request #3 from benedictkhoomw/branch-A-Lambdas

Refactor user input handler to lambda

benedictkhoomw committed:
Disable window resizability

benedictkhoomw committed:
Add shorter aliases for keywords

<KEYWORD>: <ALIAS 1>, [<ALIAS 2>]
---------------------------------
todo: t
deadline: de
event: e
delete: del, rm
done: d
bye: b
find: f
list: l

benedictkhoomw committed:
Add feedback for delete and done commands if no tasks

benedictkhoomw committed:
Merge pull request #5 from benedictkhoomw/branch-BCD-Extension

Add shorter aliases to command keywords

benedictkhoomw committed:
Fix find command accepted format message

benedictkhoomw committed:
Merge branch 'branch-BCD-Extension'

benedictkhoomw committed:
Add red text for error messages

benedictkhoomw committed:
Merge pull request #6 from benedictkhoomw/branch-A-BetterGui

Add red text for error messages

benedictkhoomw committed:
Add user guide and customize README

benedictkhoomw committed:
Add prompt text to user input box

benedictkhoomw committed:
Merge branch 'branch-A-BetterGui'

benedictkhoomw committed:
Merge branch 'branch-A-UserGuide'

benedictkhoomw committed:
Update user guide images

benedictkhoomw committed:
Merge branch 'branch-A-UserGuide'

benedictkhoomw committed:
Shrink user guide image

benedictkhoomw committed:
Merge branch 'branch-A-UserGuide'

benedictkhoomw committed:
Fix README.md image

benedictkhoomw committed:
Fix README.md URL

benedictkhoomw committed:
Merge branch 'branch-A-UserGuide'

benedictkhoomw committed:
Rename app screenshots

benedictkhoomw committed:
Merge branch 'branch-A-UserGuide'
